lib/rspec/matchers/built_in/yield.rb

```diff
@@ -154,7 +154,7 @@ def matches?(block)
           @probe = YieldProbe.probe(block)
           return false unless @probe.has_block?
 
-          @probe.num_yields.__send__(@expectation_type, @expected_yields_count)
+          @probe.num_yields.__send__(@expectation_type || :>=, @expected_yields_count || 1)
```

Comment:
Ok I'd prefer a guard that checks if these are both nil, sends the default, rather than this or, even if in theory its impossible one could be nil and the other not.

Reply:
Ok. Note: not possible for one to be `nil` and not the other (after this PR)

Reply:
Its possible, its just not supported.

Reply:
> Its possible, its just not supported.

I'm surprised by how confident you seem to be about this. How you would get one set to `nil` and the other at something else (after this PR)?

Reply:
Extensions of this class and/or modifying the internals.